packages/flutter_tools/lib/src/migrate/migrate_config.dart

```diff
@@ -0,0 +1,257 @@
+      case 'root': {
+        platformDir = project.directory;
+        break;
+      }
+      case 'android': {
+        platformDir = project.android.hostAppGradleRoot;
+        break;
+      }
+      case 'ios': {
+        platformDir = project.ios.hostAppRoot;
+        break;
+      }
+      case 'web': {
+        platformDir = project.web.directory;
+        break;
+      }
+      case 'macos': {
+        platformDir = project.macos.hostAppRoot;
+        break;
+      }
+      case 'linux': {
+        platformDir = project.linux.managedDirectory.parent;
+        break;
+      }
+      case 'windows': {
+        platformDir = project.windows.managedDirectory.parent;
+        break;
+      }
+      case 'windowsUwp': {
+        platformDir = project.windowsUwp.managedDirectory.parent;
+        break;
+      }
+      case 'fuchsia': {
+        platformDir = project.fuchsia.editableHostAppDirectory;
+        break;
+      }
+    }
+    if (platformDir == null) {
+      throwToolExit('Invalid platform when creating MigrateConfig', exitCode: 1);
+    }
+    return platformDir.childFile(kFileName);
+  }
+
+  /// Returns a list of platform names that are supported by the project.
+  static List<String> getSupportedPlatforms({bool includeRoot = false, FlutterProject? flutterProject}) {
+    final List<String> platforms = includeRoot ? <String>['root'] : <String>[];
+    flutterProject ??= FlutterProject.current();
+    if (flutterProject.android.existsSync()) {
+      platforms.add('android');
+    }
+    if (flutterProject.ios.exists) {
+      platforms.add('ios');
+    }
+    if (flutterProject.web.existsSync()) {
+      platforms.add('web');
+    }
+    if (flutterProject.macos.existsSync()) {
+      platforms.add('macos');
+    }
+    if (flutterProject.linux.existsSync()) {
+      platforms.add('linux');
+    }
+    if (flutterProject.windows.existsSync()) {
+      platforms.add('windows');
+    }
+    if (flutterProject.windowsUwp.existsSync()) {
+      platforms.add('windowsUwp');
+    }
+    if (flutterProject.fuchsia.existsSync()) {
+      platforms.add('fuchsia');
+    }
+    return platforms;
+  }
+
+  /// Finds the fallback revision to use when no base revision is found in the .migrate_config.
+  static Future<String> getFallbackBaseRevision() async {
+    // Use the .metadata file if it exists.
+    final File metadataFile = FlutterProject.current().directory.childFile('.metadata');
+    if (metadataFile.existsSync()) {
+      final FlutterProjectMetadata metadata = FlutterProjectMetadata(metadataFile, globals.logger);
+      if (metadata.versionRevision != null) {
+        return metadata.versionRevision!;
+      }
+    }
+    return _getGitHash(Cache.flutterRoot!);
+  }
+
+  static Future<String> _getGitHash(String projectPath, [String tag = 'HEAD']) async {
+    final List<String> cmdArgs = <String>['rev-parse', tag];
+    final ProcessResult result = await Process.run('git', cmdArgs, workingDirectory: projectPath);
```

Comment:
I'll be moving this into a utility class in the next PR where I handle all of this. I'd prefer to not replicate all the error handling here temporarily as it will be removed very soon.